eva-server/src/test/java/uk/ac/ebi/eva/server/ws/GeneWSServerV2Test.java

```diff
@@ -0,0 +1,164 @@
+    private static final String MAIN_ID = "rs1";
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @MockBean
+    private FeatureService featureService;
+
+    @MockBean
+    private VariantWithSamplesAndAnnotationsService variantService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private String GENE_ID1 = "ENSG00000227232";
+
+    private String GENE_ID2 = "ENSG00000227244";
+
+    @Before
+    public void setUp() throws Exception {
+        VariantWithSamplesAndAnnotation variantEntity = new VariantWithSamplesAndAnnotation("20", 1000, 1005,
+                "reference", "alternate", MAIN_ID);
+
+        List<Region> oneRegion = Collections.singletonList(new Region("20", 60000L, 62000L));
+        given(variantService.findByRegionsAndComplexFilters(eq(oneRegion), any(), any(), any(), any()))
+                .willReturn(Collections.singletonList(variantEntity));
+        given(variantService.countByRegionsAndComplexFilters(eq(oneRegion), any())).willReturn(1l);
+
+        List<Region> twoRegions = Arrays.asList(new Region("20", 60000L, 62000L),
+                new Region("20", 63000L, 64000L));
+
+
+        given(variantService.findByRegionsAndComplexFilters(eq(twoRegions), any(), any(), any(), any()))
+                .willReturn(Arrays.asList(variantEntity, variantEntity));
+        given(variantService.countByRegionsAndComplexFilters(eq(twoRegions), any())).willReturn(2l);
+
+        given(variantService
+                .findByRegionsAndComplexFilters(not(or(eq(oneRegion), eq(twoRegions))), any(), any(), any(), any()))
+                .willReturn(Collections.emptyList());
+        given(variantService.countByRegionsAndComplexFilters(not(or(eq(oneRegion), eq(twoRegions))), any()))
+                .willReturn(0l);
+
+        FeatureCoordinates feature1 = new FeatureCoordinates(GENE_ID1, "id", "feature", "20", 60000L, 62000L);
+        FeatureCoordinates feature2 = new FeatureCoordinates(GENE_ID1, "id", "feature", "20", 63000L, 64000L);
+
+        List<FeatureCoordinates> featureCoordinate = Arrays.asList(feature1);
+        List<String> geneId = Arrays.asList(GENE_ID1);
+        given(featureService.findAllByGeneIdsOrGeneNames(eq(geneId), eq(geneId))).willReturn(featureCoordinate);
+
+        List<FeatureCoordinates> featureCoordinates = Arrays.asList(feature1, feature2);
+        List<String> geneIds = Arrays.asList(GENE_ID1, GENE_ID2);
+        given(featureService.findAllByGeneIdsOrGeneNames(eq(geneIds), eq(geneIds))).willReturn(featureCoordinates);
+    }
+
+    @Test
+    public void testGetVariantsByExistingGene() throws URISyntaxException {
+        assertEquals("20", testGetVariantsGeneHelper("ENSG00000227232", 1, HttpStatus.OK).get(0));
+    }
+
+    private List<String> testGetVariantsGeneHelper(String testRegion, int expectedVariants, HttpStatus status)
+            throws URISyntaxException {
+        String url = "/v2/genes/" + testRegion + "/variants?species=mmusculus&assembly=grcm38";
```

Comment:
I have actually used  same way given above to construct URLs in Region endpoint as well. Should I change it?

Reply:
I see, I checked the V1 of regions test and not the one you created. I prefer using the builder because makes the URL more clear than a concatenated String but given that everything is implemented that way we can leave it as is.

Reply:
Okay :+1: